### PR TITLE
Fix #1523, solving inconsistency in SpotBugs Bug Description of EQ_COMPARING_CLASS_NAMES

### LINKED ISSUE #1523: Inconsistent bug description on EQ_COMPARING_CLASS_NAMES
Hi,
I found an inconsistent bug description on `EQ_COMPARING_CLASS_NAMES`.

The [description](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#eq-equals-method-compares-class-names-rather-than-class-objects-eq-comparing-class-names) mentioned that “This method checks to see if two objects are the same class by checking to see if the names of their classes are equal” so one would expect SpotBugs to give a warning in the statement if(c.getClass().getName().equals(config.getClass().getName())). However, SpotBugs did not give any warning because its implementation actually checks if the compare statement is inside the equals method according to line 336 in [OverridingEqualsNotSymmetrical.java](https://github.com/spotbugs/spotbugs/blob/51e586bed98393e53559a38c1f9bd15f54514efa/spotbugs/src/main/java/edu/umd/cs/findbugs/detect/OverridingEqualsNotSymmetrical.java#L336)

When we compare the bug description for this pattern and other related patterns, we think that the description should be
changed to “This class defines an equals method that...”

### PR DESCRIPTION
**Pull request by gloNelson**
Hi, 
Fixes #1523 
I modified the pattern `EQ_COMPARING_CLASS_NAMES` in spotbugs/etc/messages.xml to fix #1523.
Please help me review @KengoTODA, thanks :)

----

Make sure these boxes are checked before submitting your PR -- thank you!

- [x] Added an entry into `CHANGELOG.md` if you have changed SpotBugs code

### CODE REVIEW

File: docs/locale/ja/LC_MESSAGES/bugDescriptions.po [hunk 1]
```diff
@@ -12543,3 +12543,18 @@ msgstr ""
 #~ "</p>"
 #~ msgstr ""
 
+#~ msgid ""
```
Comment (ThrawnCA) [inline docs/locale/ja/LC_MESSAGES/bugDescriptions.po:12546]: Why add these lines?

**Review by ThrawnCA — DISMISSED** [review]

File: docs/locale/ja/LC_MESSAGES/bugDescriptions.po (hunk 1 above)
Comment (gloNelson) [reply docs/locale/ja/LC_MESSAGES/bugDescriptions.po:12546]: I don't know. I just modified `spotbugs/etc/messages.xml`  and executed `./build.sh` in the **doc** directory.

Comment (KengoTODA) [reply docs/locale/ja/LC_MESSAGES/bugDescriptions.po:12546]: we can safely remove these lines.

**Review by ThrawnCA — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -18,6 +18,7 @@ Currently the versioning policy of this project follows [Semantic Versioning v2.
 - False positive OBL_UNSATIFIED_OBLIGATION with try with resources ([#79](https://github.com/spotbugs/spotbugs/issues/79))
 - `SA_LOCAL_SELF_COMPUTATION` bug  ([#1472](https://github.com/spotbugs/spotbugs/issues/1472))
 - False positive `EQ_UNUSUAL` with record classes ([#1367](https://github.com/spotbugs/spotbugs/issues/1367))
+- Inconsistent bug description on `EQ_COMPARING_CLASS_NAMES` ([#1523](https://github.com/spotbugs/spotbugs/issues/1523))
```
Comment (KengoTODA) [inline CHANGELOG.md:21]: Please merge the latest `origin/master` and move this line to `## Unreleased`.

**Review by KengoTODA — CHANGES_REQUESTED** [review]

**Review by KengoTODA — APPROVED** [review]

**Pull request merged**